crates/db_schema/src/lib.rs

```diff
@@ -143,6 +143,9 @@ pub mod functions {
   sql_function! {
     fn hot_rank(score: BigInt, time: Timestamp) -> Integer;
   }
+
+  sql_function!(fn lower(x: Text) -> Text);
+
```

Comment:
This shouldn't be necessary, diesel already has a lower function, you just have to import it. https://docs.diesel.rs/diesel/query_dsl/trait.QueryDsl.html?search=lower#method.select

Reply:
I chose this method after reading a possibly oudated answer to diesel-rs/diesel#560, and [this](http://docs.diesel.rs/diesel/macro.sql_function.html) section of the manual. 

I have tried to find specifically where I should load this function from, but in their internal tests and in the examples that I have found the function is defined locally wherever it is called. 

The only place where I see the function defined under diesel:: is here: diesel/diesel/src/pg/connection/mod.rs 
It is defined to be used locally for a test and I don't know if it makes sense to import the function from there.

I have checked the source of QueryDsl and I can't find the method defined there.

Reply:
Ah my bad, I'd assumed it was one of their expression methods. That's right that you do have to add that sql_function then.